lib/iris/fileformats/grib/__init__.py

```diff
@@ -239,7 +239,7 @@ def __init__(self, grib_message, grib_fh=None, auto_regularise=True):
             # The byte offset requires to be reset back to the first byte
             # of this message. The file pointer offset is always at the end 
             # of the current message due to the grib-api reading the message.
-            proxy = GribDataProxy(shape, np.zeros(0).dtype, np.nan,
+            proxy = GribDataProxy(shape, np.array([0.]).dtype, np.nan,
```

Comment:
I still don't understand why this needs to be done rather than `np.float_` (as suggested in [this comment](https://github.com/SciTools/iris/pull/2468#issuecomment-291400403)). Surely it just needs a float datatype?